src/main/java/com/example/Restaurant_Management_System_REST_API/repository/CustomerRepository.java

```diff
@@ -1,13 +1,19 @@
 package com.example.Restaurant_Management_System_REST_API.repository;
 
 import com.example.Restaurant_Management_System_REST_API.model.entity.Customer;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
 import java.util.Optional;
 
 @Repository
 public interface CustomerRepository extends CrudRepository<Customer, Long> {
 
     Optional<Customer> findByEmailAddress(String email);
+    @Query("SELECT c FROM Customer c JOIN c.authorities a WHERE a.name = :name")
+    List<Customer> findByAuthorities_Authority_name(String name);
+    @Query("SELECT c FROM Customer c WHERE c.reservation IS NOT NULL")
+    List<Customer> findByReservation();
```

Comment:
Great usage of @Query :)

Reply:
great! I loved to do this in this branch! This was very good practice!